src/interfaces/records/handlers/records-write.ts

```diff
@@ -83,12 +77,11 @@ export class RecordsWriteHandler implements MethodHandler {
       try {
         await StorageController.put(this.messageStore, this.dataStore, tenant, incomingMessage, indexes, dataStream);
       } catch (error) {
-        if (error.code === DwnErrorCode.MessageStoreDataCidMismatch ||
-            error.code === DwnErrorCode.MessageStoreDataNotFound ||
-            error.code === DwnErrorCode.MessageStoreDataSizeMismatch) {
-          return new MessageReply({
-            status: { code: 400, detail: error.message }
-          });
+        const e = error as any; // FIXME, support code filters in MessageReply.fromError?
```

Comment:
TBH I like that the error code check is explicit here. Makes it easy to cmd+F. Adding filters to `MessageReply.fromError` seems like overkill.